packages/lexicon/src/validators/complex.ts

```diff
@@ -19,25 +19,46 @@ export function validate(
   def: LexUserType,
   value: unknown,
 ): ValidationResult {
+  // Order based on the number of occurrences in the Bluesky official lexicons:
```

Comment:
Does this have a performance impact?  I wouldn't be surprised if V8 optimizes this into a simple O(1) lookup rather than running through each case.

Reply:
I don't know that SpiderMonkey and V8 optimize switch statements into lookup tables. I kind of assumed they don't since I read [this article](https://artemis.sh/2022/08/07/emulating-calculators-fast-in-js.html) a while back. But things have probably changed since I read that.

The main point is to change a seemingly random ordering here into something that makes sense, while potentially benefiting from better performances.

Reply:
re-reading the article now (in particular its conclusion) I realize that I should always rely on my experience & memory 🙃

Reply:
https://github.com/bluesky-social/atproto/pull/2999/commits/617d86ca475c3ce68c27e694f22189c1628a4388#diff-7d6f311165a765285060eb4c8944d5b029c4a4d48ce8dd636d86b37f50644afd